app/models/driver.rb

```diff
@@ -0,0 +1,40 @@
+
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+
+  # must provide a name
+  validates :name, presence: {message: "%{value} must be given"}
+
+  # must provide a vin
+  validates :vin, presence: {message: "%{value} must be given"}
+
+  def self.select_driver
+    return Driver.all.sample
+  end
+
+  def average_rating
+    sum = 0
+    counter = 0
+    trips.each do |trip|
+      if trip.rating != nil
+        sum += trip.rating
+        counter += 1
+      end
+    end
+    if sum != 0
+      average = sprintf('%.2f', sum.to_f/counter)
```

Comment:
To maintain a clean separation between what is "model logic" and what is "view/display logic" we should actually have this method just return `sum.to_f / counter` and do the `sprintf` call in our view files.